src/tools/calculator.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    BaseToolRunOptions,
+    StringToolOutput,
+    Tool,
+    ToolInput,
+  } from "@/tools/base.js";
+  import { z } from "zod";
+  import { create, all } from 'mathjs'
+  
+  const math = create(all)
+  const limitedEvaluate = math.evaluate
+  math.import({
+    // most important (hardly any functional impact)
+    'import':     function () { throw new Error('Function import is disabled') },
+    'createUnit': function () { throw new Error('Function createUnit is disabled') },
+    'reviver':    function () { throw new Error('Function reviver is disabled') },
+  
+    // extra (has functional impact)
+    'evaluate':   function () { throw new Error('Function evaluate is disabled') },
+    'parse':      function () { throw new Error('Function parse is disabled') },
+    'simplify':   function () { throw new Error('Function simplify is disabled') },
+    'derivative': function () { throw new Error('Function derivative is disabled') },
+    'resolve':    function () { throw new Error('Function resolve is disabled') },
+  }, { override: true })
+
+  export class CalculatorTool extends Tool<StringToolOutput> {
```

Comment:
I would include a disclaimer/warning that are are potential security and stability risks involved in using `.evaluate`, even with the limited version. Pointing to the mathjs [documentation](https://mathjs.org/docs/expressions/security.html).

@Tomas2D This is likely not the first time we would be introducing a potentially dangerous tool. Any specific guidance on how approach this (comment at the top, having a specific flag turned on/off)? For example check the request tool from [langchain.](https://github.com/langchain-ai/langchain/blob/34fc00aff17d1913a1cfe4b205e732982f98f7b8/libs/community/langchain_community/tools/requests/tool.py#L36). I think users should know that this is not safe tool unless run in isolation. Another approach could be to add `unsafe` suffix to the tool name for all tools we cannot guarantee safeness OOTB.

Reply:
I agree that there should be a disclaimer.

Reply:
If we decide to use the flag, we would need to add it to other tools, like `WebCrawlerTool`.

Reply:
I added a warning re. security and stability of the tool, linking to the mathjs documentation.